be/src/olap/rowset_graph.cpp

```diff
@@ -237,25 +232,26 @@ OLAPStatus RowsetGraph::capture_consistent_versions(const Version& spec_version,
         reversed_path.push_back(tmp_vertex_index);
     }
 
-    // Make version_path from reversed_path.
-    std::stringstream shortest_path_for_debug;
-    for (size_t path_id = reversed_path.size() - 1; path_id > 0; --path_id) {
-        int64_t tmp_start_vertex_value = _version_graph[reversed_path[path_id]].value;
-        int64_t tmp_end_vertex_value = _version_graph[reversed_path[path_id - 1]].value;
-
-        // tmp_start_vertex_value mustn't be equal to tmp_end_vertex_value
-        if (tmp_start_vertex_value <= tmp_end_vertex_value) {
-            version_path->emplace_back(tmp_start_vertex_value, tmp_end_vertex_value - 1);
-        } else {
-            version_path->emplace_back(tmp_end_vertex_value, tmp_start_vertex_value - 1);
-        }
+    if (version_path != nullptr) {
```

Comment:
I think should not coupling outside VersionPath empty logic in this function.

Reply:
I found `RowsetGraph::capture_consistent_versions` is now only called by `Tablet::capture_consistent_versions`, the latter is now called by two purposes. First, check version integrity, second, check version integrity and then capture versions when check success. For the first purpose, it's not needed to construct and fill a output parameter. Of course the function name `capture_consistent_versions ` maybe not accuracy in this case.

Reply:
Either `version_path`  is nullptr or is valid for this function. If it's nullptr, just validate rowset graph; if it's not nullptr, validate rowset graph and return current versions.